weatherman.py

```diff
@@ -0,0 +1,208 @@
+        
+    def track_min_temperature(self, temperature, date): 
+        if temperature != "":
+            self.min_temperature = min(self.min_temperature, int(temperature))
+            if self.min_temperature == int(temperature): self.min_temperature_date = date
+
+    def track_max_humidity(self, humidity, date):
+        if humidity != "":
+            self.max_humidity = max(self.max_humidity, int(humidity))
+            if self.max_humidity == int(humidity): self.max_humidity_date = date
+
+    def print_temperatures_year(self):
+        print(f"highest temperature is: {self.max_temperature} C on {self.max_temperature_date}")
+        print(f"lowest temperature is: {self.min_temperature} C on {self.min_temperature_date}")
+        print(f"highest humidity percentage is: {self.max_humidity} % on {self.max_humidity_date}")
+
+
+class WeatherFilesHandler:
+
+    def __open_csv_file(self, file_name):    
+        csv_file = open(file_name) 
+        return csv_file
+
+    def __read_weather_data_from_csv(self, csv_file, file_name):
+        weather_data = list(csv.DictReader(csv_file, skipinitialspace=True))
+        if len(weather_data) != 0: return weather_data
+        print(f"No Record Found against file : {file_name}")
+    
+    def return_weather_data(self, file_name):
+        csv_file = self.__open_csv_file(file_name)
+        weather_data = self.__read_weather_data_from_csv(csv_file, file_name)
+        return weather_data
+
+    def return_all_file_names_list_for_year(self, year):
+        file_names=[]
+        flag = False
+        for file_name in os.listdir('.'):
+            if year in file_name:
+                file_names.append(file_name)
+                flag = True
+        return file_names if flag else 0
+    
+    @staticmethod
+    def match_and_return_file_name(year, month_name):      
+        flag = False    
+        for file_name in os.listdir('.'):
+            if year in file_name and month_name in file_name:
+                return file_name          
+        if not flag:     
+            print("NO RECORD FOUND AGAINT THIS NAME")
+            return False
+
+
+class WeatherDataCalculator:
+
+    def __calculate_average(self, sum_of_temperature_values, total_number_of_values):    
+        for key in sum_of_temperature_values:                           
+            sum_of_temperature_values[key] = sum_of_temperature_values[key] / total_number_of_values
+        return sum_of_temperature_values
+
+    def __extract_int_value(self, value):
+        return int(value) if value != "" else 0
+    
+    def __calculate_temprature_sum_for_each_record(self, weather_data, sum_of_temperature_values, key): 
+        for line_read in weather_data:                                    
+                sum_of_temperature_values[key] += self.__extract_int_value(line_read[key])            
+        return  sum_of_temperature_values[key]
+    
+    def calculate_temperature_sum_and_return_average(self, weather_data):
```

Comment:
When function is actually returning average so it means that average calculation is being made. Average has to include sum in any case so naming it in this way looks bad.